src/Features/CSharp/Portable/ConvertNamespace/ConvertNamespaceDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,86 @@
+            EnforceOnBuildValues.UseFileScopedNamespace,
+            new LocalizableResourceString(nameof(CSharpFeaturesResources.Convert_to_file_scoped_namespace), CSharpFeaturesResources.ResourceManager, typeof(CSharpFeaturesResources)));
+
+        private static readonly ImmutableDictionary<DiagnosticDescriptor, ILanguageSpecificOption> s_descriptors =
+            ImmutableDictionary<DiagnosticDescriptor, ILanguageSpecificOption>.Empty
+                .Add(s_useRegularNamespaceDescriptor, CSharpCodeStyleOptions.PreferFileScopedNamespace)
+                .Add(s_useFileScopedNamespaceDescriptor, CSharpCodeStyleOptions.PreferFileScopedNamespace);
+
+        public ConvertNamespaceDiagnosticAnalyzer()
+            : base(s_descriptors, LanguageNames.CSharp)
+        {
+        }
+
+        public sealed override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterSyntaxNodeAction(AnalyzeNamespace, SyntaxKind.NamespaceDeclaration, SyntaxKind.FileScopedNamespaceDeclaration);
+
+        private void AnalyzeNamespace(SyntaxNodeAnalysisContext context)
+        {
+            var options = context.Options;
+            var namespaceDeclaration = (BaseNamespaceDeclarationSyntax)context.Node;
+            var syntaxTree = namespaceDeclaration.SyntaxTree;
+
+            var cancellationToken = context.CancellationToken;
+            var root = (CompilationUnitSyntax)syntaxTree.GetRoot(cancellationToken);
+
+            var optionSet = options.GetAnalyzerOptionSet(syntaxTree, cancellationToken);
+
+            var diagnostic = AnalyzeNamespace(optionSet, root, namespaceDeclaration);
+            if (diagnostic != null)
+                context.ReportDiagnostic(diagnostic);
+        }
+
+        private static Diagnostic? AnalyzeNamespace(OptionSet optionSet, CompilationUnitSyntax root, BaseNamespaceDeclarationSyntax declaration)
+        {
+            var tree = declaration.SyntaxTree;
+            var preferFileScopedNamespace = optionSet.GetOption(CSharpCodeStyleOptions.PreferFileScopedNamespace);
+            var severity = preferFileScopedNamespace.Notification.Severity;
+
+            var descriptor =
+                ConvertNamespaceHelper.CanOfferUseRegular(optionSet, declaration, forAnalyzer: true) ? s_useRegularNamespaceDescriptor :
+                ConvertNamespaceHelper.CanOfferUseFileScoped(optionSet, root, declaration, forAnalyzer: true) ? s_useFileScopedNamespaceDescriptor : null;
+            if (descriptor == null)
+                return null;
+
+            // if the diagnostic is hidden, show it anywhere from the `namespace` keyword through the name.
+            // otherwise, if it's not hidden, just squiggle the name.
+            var diagnosticLocation = severity.WithDefaultSeverity(DiagnosticSeverity.Hidden) == ReportDiagnostic.Hidden
```

Comment:
I'm worried the `severity` variable here only reflects the `option = value:severity` syntax but not reflecting the `dotnet_diagnostic.ID.severity = severity` syntax.

Reply:
so this follows what we do with use-expression-body.  i'm going to stick with this pattern under teh presumption that it's correct.  if not, we shoudl fix it for all of these.  i do not want to have multiple different patterns at the same time :)

Reply:
Tagging @mavasani to confirm if there is a problem here in which case I'll file an issue.